Sending email to landlord & tenant

anshulsahni opened the pull request:
- [PACKAGE]:added nodemailer package
- [EMAIL]: Added EmailService class
- Sending email to landlord & tenant in BookService

anshulsahni committed:
[PACKAGE]:added nodemailer package

anshulsahni committed:
[EMAIL]: Added & implemented email service

	- Added EmailService class
	- Sending optional email to tenant & landlord in Book Service

tarunbatra commented on finch/src/services/email.js:
```diff
@@ -0,0 +1,51 @@
+
+  setSender(senderEmail) {
+    this.from = senderEmail;
+    return this;
```
👍 for the fluent design here!

tarunbatra commented on finch/src/services/email.js:
```diff
@@ -0,0 +1,51 @@
+    this.sender = emailSender;
+  }
+
+  init() {
```
why shouldn't this be merged with the constructor?

anshulsahni replied:
I'm also thinking the same now, can't think of a reason of why this shouldn't be part constructor

tarunbatra commented on finch/src/services/email.js:
```diff
@@ -0,0 +1,51 @@
+  }
+
+  init() {
+    this.transporter = nodemailer.createTransport({
```
you my want to check here if `this.transporter` exists already.

anshulsahni replied:
I guess the need for checking if `this.transporter` exist or not will be removed if we include this initialization in constructor only

tarunbatra commented on finch/src/services/email.js:
```diff
@@ -0,0 +1,51 @@
+  }
+
+  async send() {
+    const { transporter, from, to, html } = this;
```
nitpick: probably its a good idea to throw an error here if one of these parameters don't exist.

anshulsahni replied:
actually not a nitpick but a very good idea, this would save us from failing miserably while sending the email

anshulsahni committed:
[REFACTOR][EMAIL]: Changed the API for services/email

	- removed init & moved it's code to constructor
	- Sender of email is not by default company sender

anshulsahni commented:
@tarunbatra done will the changes suggested by you, please check

tarunbatra reviewed (approved):
LGTM 👍

anshulsahni committed:
Merge branch 'main' into send_email_integration

anshulsahni committed:
🐞fix(config): fixed importing of <appEnv>.cjs

anshulsahni committed:
🩺test(emailservice): added tests

anshulsahni committed:
🩺test(emailservice): tests for validateEmailParams

anshulsahni committed:
🧹lint(books): linting fixed in service
